go.mod

```diff
@@ -24,14 +24,14 @@ go 1.21
 require (
 	github.com/apache/yunikorn-scheduler-interface v0.0.0-20231211235204-ec7bfad7d00e
 	github.com/google/btree v1.1.2
-	github.com/google/go-cmp v0.5.9
-	github.com/google/uuid v1.3.0
+	github.com/google/go-cmp v0.6.0
+	github.com/google/uuid v1.6.0
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/looplab/fsm v1.0.1
 	github.com/opentracing/opentracing-go v1.2.0
-	github.com/prometheus/client_golang v1.13.1
-	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.37.0
+	github.com/prometheus/client_golang v1.16.0
```

Comment:
This jumps over a major incompatible version 0.14. Upgrade is logged as  YUNIKORN-2394.

Reply:
oh I did not notice that incompatible change. Big thanks for reminder!

Reply:
already rebased to the latest commit and addressed comments, huge thanks ! 😃